avocado/utils/service.py

```diff
@@ -775,15 +775,12 @@ class has methods to start/stop services.
     # service_command_generator is explicitly undefined so that we get the
     # NameError on first access
     # pylint: disable=W0601
-    global _service_manager
-    try:
-        return _service_manager
-    except NameError:
-        internal_generator = _get_service_command_generator
-        internal_parser = _get_service_result_parser
-        _service_manager = internal_service_manager(internal_generator(run),
-                                                    internal_parser(run))
```

Comment:
nitpick: it'd be nice to split the bugfixes and first address the `run=run` issue along with description and then the caching issue as a separate commits. It might look unnecessary to you, but it makes it easier to follow and see what was actually fixed as well as it makes it actually possible to revert the actual little pieces should some problem arise.

Reply:
Thank you for the split. There is a small issue with the commit message as it claims this fixes the `multi-access` issue, which is not true. This only makes sure the runner is not omitted when creating the service class for the first time, the next commit fixes the multi-access issue.

Also please configure your git to wrap lines.

Reply:
@ldoktor IMHO commit message is ok. There is no `multi-access` word in it. Maybe you mixed it up with https://github.com/avocado-framework/avocado/pull/2640/commits/cec424f477357146c7a6074259cf7e075fd5b513. Will correct wrapping..

Reply:
When you apply this patch only and run the 2 commands mentioned in the card you claim it fixes it will fail. Therefor it is not addressing issue from that card...

Reply:
Well, simply add a selftest and you'll see ;-)